airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplier.java

```diff
@@ -35,25 +41,37 @@ public OAuthConfigSupplier(ConfigRepository configRepository, boolean maskSecret
   public JsonNode injectSourceOAuthParameters(UUID sourceDefinitionId, UUID workspaceId, JsonNode sourceConnectorConfig)
       throws IOException {
     try {
+      final ImmutableMap<String, Object> metadata = generateSourceMetadata(sourceDefinitionId);
       // TODO there will be cases where we shouldn't write oauth params. See
       // https://github.com/airbytehq/airbyte/issues/5989
       MoreOAuthParameters.getSourceOAuthParameter(configRepository.listSourceOAuthParam().stream(), workspaceId, sourceDefinitionId)
           .ifPresent(
-              sourceOAuthParameter -> injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration()));
+              sourceOAuthParameter -> {
+                injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration());
+                if (!maskSecrets) {
+                  TrackingClientSingleton.get().track(workspaceId, "OAuth Injection - Backend", metadata);
```

Comment:
Should this be tracked here or only during syncs? (ignore checkConnections and discover)

Reply:
It seems useful to track it everywhere for debugging/lineage purposes

also I think we should make it a little clearer in a comment why we only track when masking secrets (probably because we're not actually injecting anything?)